libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -103,6 +106,7 @@ proc getPeer(p: PubSub,
 
   p.peers[peer.id] = peer
   peer.refs.inc # increment reference cound
+  peer.observers = p.observers
```

Comment:
Why `PubSubPeer` object needs to have observers?

Reply:
In order to intercept `encode` without re-implementing everything.